app/uk/gov/hmrc/cardpaymentfrontend/forms/EmailAddressForm.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cardpaymentfrontend.forms
+
+import play.api.data.Forms.mapping
+import play.api.data.format.Formatter
+import play.api.data.{Form, FormError, Forms}
+import uk.gov.hmrc.cardpaymentfrontend.models.EmailAddress
+
+import scala.util.matching.Regex
+
+object EmailAddressForm {
+
+  val emailAddressKey: String = "email-address"
+
+  private val emailAddressRegex: Regex = """^[a-zA-Z0-9\.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$""".r
+
+  def form(): Form[EmailAddress] = {
+    val emailAddressMapping = Forms.of(new Formatter[EmailAddress]() {
+
```

Comment:
This approach differs slightly from how it was done in pay-frontend. I chose this way in favour of keeping it within the form method and I _think_ it's a bit simpler, i.e. just a few if else's instead of using constraints etc. Happy to tweak this if the consensus is the constraint implementation is preferred.

TL;DR: Slightly different to pay-frontend implementation, but functionally the same

Reply:
Will this form ever b e referenced outside of the Controller? Should we consider keeping it _with_ the Controller?

Reply:
Probably not, but my opinion is having it in it's own object keeps the controller code cleaner/simpler. I suppose we could put it into a companion object for the controller. I'm not fussed either way.